utils/file.js

```diff
@@ -76,14 +78,16 @@ exports.fileSystem = (function(){
 
       while (null !== (chunk = readable.read(chunkSize))) {
         const chunkId = sha1HashData(chunk);
-        manifest.chunks.push(chunkId);
+        manifest.chunks[chunkId] = [];
 
+        copyShards(chunk, chunkId, manifest)
```

Comment:
Since this method is creating shard ids for duplicate shards and writing them to the manifest, the name could also be `createCopyIdsforRedundantShardsAndWriteToManifest`. I remember Brandon told me if I ever have to use an and in my method, I'm doing too much. What do you think about breaking it down a bit more into something like this?

```
const redudantShardIds = createRedundantShardIds(chunk, chunkId) 
// you could also use the word copy instead of redundant here (i.e copyShardIds) - 
// but that almost sounds like a verb/action/function name to me so I went with redundant
manifest.chunks[chunkId] = [...redundantShardIds]
```

Reply:
Yeah, I agree that copyShardIds might be better. Write manifest will be done after the read stream finished so write to manifest won't apply to this method.